packages/frontend/src/components/composer/Composer.tsx

```diff
@@ -129,6 +138,41 @@ const Composer = forwardRef<
     ? editMessageInputRef
     : regularMessageInputRef
 
+  if (useHasChanged2(chatId) && recording) {
+    setRecording(false)
+  }
+
+  const saveVoiceAsDraft = (voiceData: Blob) => {
+    const reader = new FileReader()
+    reader.readAsDataURL(voiceData)
+    reader.onloadend = async () => {
+      if (!reader.result) {
+        log.error('Cannot convert blob to base64. reader.result is null')
+        return
+      }
+      const b64 = reader.result.toString().split(',')[1]
+      // random filename
+      const filename = Math.random().toString(36).substring(2, 10) + '.mp3'
+      const path = await runtime.writeTempFileFromBase64(filename, b64)
+      addFileToDraft(path, basename(path), 'Voice').catch((reason: any) => {
```

Comment:
`basename` appears to do nothing on Windows. It uses Browserify polyfill. Why don't we use `filename` here?